src/components/history/GeneratedProductHistory.tsx

```diff
@@ -1,16 +1,28 @@
 import { Accordion, Alert, Badge, Button, Card, Col, ListGroup, OverlayTrigger, Row, Spinner, Table, Tooltip } from "react-bootstrap";
 import { useAppDispatch, useAppSelector } from "../../redux/hooks";
-import { GeneratedProduct, StatusTypes } from "../../types/constantTypes";
-import { useState } from "react";
+import { GeneratedProduct, StatusTypes, getUserProductsResponse } from "../../types/constantTypes";
+import { useEffect, useState } from "react";
 import { Download, Clipboard, InfoCircle } from "react-bootstrap-icons";
 import { setCurrentPage } from "../app/appSlice";
 import { generatedProductsLabels, infoIconsToRender, parameterHelp } from "../../constants/rasterParameterConstants";
+import { Product } from "../../types/graphqlTypes";
+import { getUserProducts } from "../../user/userData";
 
 const GeneratedProductHistory = () => {
-    const generatedProducts = useAppSelector((state) => state.product.generatedProducts)
+    // const generatedProducts = useAppSelector((state) => state.product.generatedProducts)
     const colorModeClass = useAppSelector((state) => state.navbar.colorModeClass)
     const dispatch = useAppDispatch()
+    const [userProducts, setUserProducts] = useState<Product[]>([])
     
+    useEffect(() => {
+        const fetchData = async () => {
+            const userProductsResponse: getUserProductsResponse = await getUserProducts()
+            if (userProductsResponse.status === 'success') setUserProducts(userProductsResponse.products as Product[])
```

Comment:
The status field is an enum with the values defined here: https://github.com/podaac/swodlr-api/blob/ba5a919e90efcbf09c8ba69914c53244574424a9/src/main/java/gov/nasa/podaac/swodlr/status/State.java#L4

Instead of hardwiring `'success'` we should be using string enums

Reply:
Got it, I will change that now

Reply:
That is actually for a general qraphql response and how I handle them vs the product status response for user's products.